src/main/java/com/github/cameltooling/lsp/internal/parser/CamelKYamlDSLParser.java

```diff
@@ -0,0 +1,150 @@
+
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+import org.yaml.snakeyaml.Yaml;
+
+import java.util.List;
+import java.util.Map;
+
+import com.github.cameltooling.lsp.internal.instancemodel.CamelURIInstance;
+import com.github.cameltooling.lsp.internal.instancemodel.YamlDSLModelHelper;
+
+/**
+ * @author Lars Heinemann
+ */
+public class CamelKYamlDSLParser extends ParserFileHelper {
+	
+	public static final String URI_KEY = "uri";
+	public static final String REST_KEY = "rest";
+	public static final String FROM_KEY = "from";
+	public static final String TO_KEY = "to";
+
+	@Override
+	public String getCamelComponentUri(String line, int characterPosition) {
+		String camelComponentURI = null;
+		Map<?, ?> data = parseYaml(line);
+		if (data.containsKey(URI_KEY) && URI_KEY.length() < characterPosition) {
+			camelComponentURI = getNonNullValue(data.get(URI_KEY));
+		} else if(data.containsKey(TO_KEY) && data.get(TO_KEY).toString().trim().length()>0) {
+			camelComponentURI = getNonNullValue(data.get(TO_KEY));
+		}
+		return camelComponentURI;
+	}
+
+	@Override
+	public CamelURIInstance createCamelURIInstance(TextDocumentItem textDocumentItem, Position position,
+			String camelComponentUri) {
+		CamelURIInstance uriInstance = new CamelURIInstance(repairLostEscapeChars(camelComponentUri), new YamlDSLModelHelper(getCorrespondingType(textDocumentItem, position.getLine())), textDocumentItem);
+		int start = getStartCharacterInDocumentOnLinePosition(textDocumentItem, position);
+		uriInstance.setStartPositionInDocument(new Position(position.getLine(), start));
+		uriInstance.setEndPositionInDocument(new Position(position.getLine(), start+repairLostEscapeChars(camelComponentUri).length()));
+		return uriInstance;
+	}
+
+	private String getNonNullValue(Object o) {
+		if (o != null) {
+			return o.toString();
+		}
+		return "";
+	}
+
+	private int getStartCharacterInDocumentOnLinePosition(TextDocumentItem textDocumentItem, Position position) {
+		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
+		String uri = extractUriFromYamlData(line);
+		if (uri.length()==0) {
+			// empty uri
+			return findStartPositionOfEmptyURI(line);
+		}
+		uri = repairLostEscapeChars(uri);
+		return line.indexOf(uri);
+	}
+
+	@Override
+	public int getPositionInCamelURI(TextDocumentItem textDocumentItem, Position position) {
+		String line = parserFileHelperUtil.getLine(textDocumentItem, position.getLine());
+		return position.getCharacter() - findStartPositionOfEmptyURI(line);
+	}
+
+	private String repairLostEscapeChars(String line) {
+		String[] parts = line.split("\"");
+		String res = "";
+		for (int i = 0; i < parts.length; i++) {
+			if (i>0) {
+				res += "\\\"";
+			}
+			res += parts[i];
+		}
+		return res;
+	}
+
+	private int findStartPositionOfEmptyURI(String line) {
+		int startPos = line.indexOf(':')+2;
+		if (line.indexOf("'", startPos) != -1) {
```

Comment:
better to use line.indexOf with char for faster performance as mentioned by SonarLint